toolchain/check/testdata/impl/lookup/min_prelude/specialization.carbon

```diff
@@ -251,6 +217,69 @@ fn F() {
   let a: C(D(E), D(E)).(Z.X) = {} as C(E, E);
 }
 
+// --- specialed_class_with_symbolic_facet_value_param.carbon
+
+interface Z {
+  let X:! type;
+}
+
+impl forall [T:! type] T as Z where .X = T {}
+
+interface Y {}
+class C(T:! Y) {}
+
+class D {}
+impl D as Y {}
+
+// D can be either a concrete or symbolic FacetValue, depending on what the
+// caller has.
+impl forall [D:! Y] C(D) as Z where .X = () {}
+
+fn F[D:! Y](d: D) {
+  // The FacetValue deduced for the param of `C` will be a symbolic FacetValue
+  // because we are in a generic where `D` is an unknown type, which will cause
+  // the query and impl self type to be C(FacetValue) for a symbolic FacetValue.
+  let a: C(D).(Z.X) = ();
```

Comment:
Because the `impl forall [D:! Y] ...` is not final, I think `C(D).(Z.X)` here should be symbolic and should not resolve to `()`. But we have no support for `final impl`s yet, so maybe add a TODO?

I think the case where we pick the impl eagerly despite it not being final is when the query is concrete. In that case we can't delay the query until the enclosing generic is instantiated (because there isn't one), so instead we'd commit early to using the parameterized impl and we'd need to reject if a more specific impl is given later.